### ISSUE TITLE
Separation of backend library code from app code

### ISSUE THREAD

**Issue by justvanrossum** [googlefonts/fontra#548]
Some of Fontra's Python code for reading/writing font files is becoming useful on its own.

To make that part of the code more accessible, it should be available with a less restrictive license, such as Apache.

This applies mostly to:
- https://github.com/googlefonts/fontra/tree/main/src/fontra/backends

The same goes for the entirety of fontra-rcjk:
- https://github.com/googlefonts/fontra-rcjk

**Comment by davelab6** [comment]
> Perhaps we should make a new package called fontra-core, that contains all library code.

I don't think that will work for people who want separate pieces; I would expect each logical unit of code to be its own pypi package. The "mono library" design of fonttools is not widely loved.

If this stuff doesn't make sense in smaller pieces, I am against it, because the Black Foundry strategy for using GPL is to encourage collaboration on the project. 

Using Fontra's Python code to read/write source data (designspace/ufo/rcjk) outside of Fontra itself in build workflows, doesn't seem to me that GPL gets in the way, since build workflows aren't distributed to end users. Using GPL code in private is fine.

**Comment by justvanrossum** [comment]
> I don't think that will work so well technically for people who want separate pieces; I would expect each logical unit of code to be its own pypi package. The "mono library" design of fonttools is not widely loved.

FWIW, the _intention_ is for the separate pieces to be logical units of code, and I agree that should be the driving force. But if indeed the license imposes to issues, then I will not rush into such a refactoring.

I do eventually want the Fontra filesystem backend modules (Python) to be installable without pulling in all Fontra's front end stuff (JS).

**Comment by davelab6** [comment]
Sounds good

I guess we can keep this issue open and revisit when there is a unit of code that would make sense as an isolated pypi package? 

The other thing about the copyleft is code under copyleft licenses can be mixed with code under non copyleft licenses that are also libre, and the non copyleft parts remain non copyleft, but the whole thing can't be made proprietary; for that, the copyleft pieces need to be replaced. So if the rest of the other project is mit/apache/etc, it can be fine to add fontra originated code.

**Comment by justvanrossum** [comment]
The question comes back with https://github.com/BlackFoundryCom/fontra-compile/issues/1.

I think I should spec out fontra-core, and we should go for it.

**Comment by rsheeter** [comment]
> The "mono library" design of fonttools is not widely loved

fwiw I have heard the exact opposite claimed as well. I don't claim to know the correct answer :)
